### PR TITLE
feat: [PL-20720]: Add following events for instrumentation

### PR DESCRIPTION
**Pull request by tiwaritejas**
- Added telemetry calls on the finish of project/organization/connector creation.
- All these calls are made from the service layer for each of these entities.

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)

### CODE REVIEW

**Comment by tiwaritejas** [comment]
trigger authorcheck

**Comment by tiwaritejas** [comment]
trigger messagecheck

**Comment by tiwaritejas** [comment]
trigger authorcheck

**Comment by tiwaritejas** [comment]
trigger authorcheck

File: 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java [hunk 1]
```diff
@@ -184,6 +188,9 @@ private ConnectorResponseDTO createInternal(
                 connectorHeartbeatTaskId.getId());
           }
         }
+        CompletableFuture.runAsync(()
+                                       -> instrumentationHelper.sendConnectorCreationFinishedEvent(
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java:192]: rename to `sendConnectorCreateEvent` ?

File: 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java [hunk 2]
```diff
@@ -371,11 +378,17 @@ public boolean delete(
               getConnectorService(connector.getType())
                   .delete(accountIdentifier, orgIdentifier, projectIdentifier, connectorIdentifier);
           if (!isDefaultBranchConnector) {
+            CompletableFuture.runAsync(()
+                                           -> instrumentationHelper.sendConnectorDeletionEvent(orgIdentifier,
+                                               projectIdentifier, connectorIdentifier, accountIdentifier));
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java:383]: sendConnectorDeleteEvent

File: 120-ng-manager/src/main/java/io/harness/ng/core/impl/ProjectServiceImpl.java [hunk 3]
```diff
@@ -159,6 +164,9 @@ public Project create(@AccountIdentifier String accountIdentifier, String orgIde
       setupProject(Scope.of(accountIdentifier, orgIdentifier, projectDTO.getIdentifier()));
       log.info(String.format("Project with identifier %s and orgIdentifier %s was successfully created",
           project.getIdentifier(), projectDTO.getOrgIdentifier()));
+      CompletableFuture.runAsync(
+          () -> instrumentationHelper.sendProjectCreationFinishedEvent(createdProject, accountIdentifier));
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/impl/ProjectServiceImpl.java:168]: same as above

File: 120-ng-manager/src/main/java/io/harness/telemetry/helpers/ConnectorInstrumentationHelper.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                .build(),
+            Category.COMMUNITY);
+        log.info("Connector Creation Finished event sent!");
+      } else {
+        log.info("There is no Account found!. Can not send Connector Creation Finished event.");
+      }
+    } catch (Exception e) {
+      log.error("Platform SendConnectorCreationFinishedEvent execution failed.", e);
+    } finally {
+      log.info("Platform SendConnectorCreationFinishedEvent execution finished.");
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/telemetry/helpers/ConnectorInstrumentationHelper.java:61]: Lets discuss on the logs

**Review by zhuoyin00 — COMMENTED** [review]
> Reviewed 2 of 13 files at r1, 11 of 11 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 13 file(s) to review. 0 of 2 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/telemetry/helpers/ProjectInstrumentationHelper.java, line 54 at r3](https://reviewable.io/reviews/harness/harness-core/30661#-Mwb4XUG-riVbPP7lIZU:-Mwb4XUH38YudoPZk0vP:b-2pylal) ([raw file](https://github.com/harness/harness-core/blob/41a7502dae32b1dffc272b6b788dc80f10c450cd/120-ng-manager/src/main/java/io/harness/telemetry/helpers/ProjectInstrumentationHelper.java#L54)):*
> ```Java
>                 .put(Destination.ALL, false)
>                 .build(),
>             Category.COMMUNITY);
> ```

Are you going to send the event for both SAAS and COMMUNITY?
If so you'll need to use sendForCommunity e.g.,
telemetryReporter.sendGroupEvent("testaccountId", null, properties, null, null,
        TelemetryOption.builder().sendForCommunity(true).build());

Same for other places.

Details https://harness.atlassian.net/wiki/spaces/GTM/pages/1586593793/Telemetry+Framework+Design+Adoption#How-to-send-data-in-Community-version

**Review by zhuoyin00 — APPROVED** [review]
> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by meenakshi-harness — COMMENTED** [review]
> Reviewed 2 of 13 files at r1, 2 of 11 files at r3, 17 of 17 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 17 file(s) to review. 0 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661#-:-Mx-FWiy9Y7Gd3sVPZ8b:bnfp4nl)* status: 17 file(s) to review. 1 of 2 approvals obtained.

**Review by zhuoyin00 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 17 file(s) to review.

**Review by meenakshi-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 17 file(s) to review.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 2 of 13 files at r1, 2 of 11 files at r3, 17 of 17 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: Some of the checks are too old.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 11 of 11 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 11 file(s) to review.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 4 of 11 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: 7 file(s) to review.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 7 of 11 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30661)* status: :shipit: complete! ✓

**Pull request merged**